### ISSUE TITLE
Adding nightly build on jenkins to make sure working with different options?

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cmake-buildscripts#74]
As you can see that adding more options will add more trouble to test stuff. I suggest to collect the popular commands (some of them were reported in this repo from me) and run them nightly, just make sure you're able to "make install".

I know it's more fun to add new stuff but building a good test systems would help us saving lots of time. cheers.

**Comment by multiplemonomials** [comment]
I agree with you, but note that with the new experimental-branch changes, you can now built Serial, MPI, OpenMP, and CUDA at the same time, and that's how I have it set up on Jenkins and on my dev computers.  Also there are some differences between the 3 different build machines that cover more options (MKL vs internal netlib, system Python vs Miniconda

I suppose the options I don't test in Jenkins are:
- `USE_FFT=FALSE`
- CMake 3.1 (the oldest version I support) <Travis used to cover this>
- CUDA (it's on my desktop but not Jenkins)
- Intel, PGI, and Clang compilers
- `BUILD_DEPRECATED=TRUE`
- The other Sander variants (LIO, APBS, PUPIL, PLUMED)

**Comment by hainm** [comment]
nice.

> you can now build Serial, MPI, OpenMP, and CUDA at the same time

as default or having some flag for that?

**Comment by multiplemonomials** [comment]
This is the default.  Passing `-DMPI=TRUE` causes all MPI executables to build both MPI and normal versions at the same time.  So, you should never need to build amber multiple times to get all the features you want.

**Comment by hainm** [comment]
I meant if user specifies "cmake amber", would MPI and openMP will be installed? 

If yes, I am not sure it's a good idea.

**Comment by multiplemonomials** [comment]
Oh, I see what you mean.  No it's not the default.  They still have to pass `-I DOPENMP=TRUE` and `-DMPI==TRUE` for OpenMP and MPI to be enabled.
